util/utility.py

```diff
@@ -492,16 +490,14 @@ def check_rewards(initiative_id: str,
             for expected_reward in expected_rewards:
                 is_rewarded = False
                 for r in actual_rewards:
-                    if r['iban'] == expected_reward.iban and r['status'] == exptected_status:
-                        total_merchant_rewards += r['amount']
-                        is_present = True
+                    if (r['iban'] == expected_reward.iban and r['status'] == exptected_status
+                            and r['amount'] == expected_reward.amount):
```

Comment:
@Benedetta-fabbri in this case if the reward is duplicated `is_rewarded` would be set to `True` two times without making the test fail.

Reply:
Resolved with new assert: `assert not is_rewarded`